pkg/operator/ceph/cluster/controller.go

```diff
@@ -384,6 +384,13 @@ func (c *ClusterController) reconcileCephCluster(clusterObj *cephv1.CephCluster,
 		return nil
 	}
 
+	if clusterObj.Status.Cephx == nil {
```

Comment:
I realized while working through some OSD stuff that this is going to be nil for both greenfield and brownfield deployments. We will need to base the init on more info than just nil I think. We can discuss more in huddles also